diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 64d8e5b..ba52fd4 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -3,7 +3,6 @@ cmake_minimum_required(VERSION 3.16...4.3)
 set(CMAKE_EXPORT_COMPILE_COMMANDS true)
 set(PYBIND_LIBNAME _pybmad)

-find_package(Python REQUIRED COMPONENTS Interpreter Development.Module)
 find_package(pybind11 CONFIG REQUIRED)

 file(GLOB GENERATED_SOURCES
@@ -11,18 +10,12 @@ file(GLOB GENERATED_SOURCES
     "src/generated/*.cpp"
 )

-python_add_library(${PYBIND_LIBNAME} MODULE
+pybind11_add_module(${PYBIND_LIBNAME} MODULE
     ${GENERATED_SOURCES}
     src/arrays.cpp
     src/common_structs.cpp
 )

-# Hide all symbols except the pybind11 module init
-set_target_properties(${PYBIND_LIBNAME} PROPERTIES
-    CXX_VISIBILITY_PRESET hidden
-    VISIBILITY_INLINES_HIDDEN ON
-)
-
 # Include directories
 target_include_directories(${PYBIND_LIBNAME} PRIVATE
     "${CMAKE_CURRENT_SOURCE_DIR}/include"
@@ -37,7 +30,6 @@ target_link_directories(${PYBIND_LIBNAME} PRIVATE "$ENV{ACC_ROOT_DIR}/${BUILD_TY

 # Link libraries
 target_link_libraries(${PYBIND_LIBNAME} PRIVATE
-    pybind11::headers
     cppbmad
     bmad
     sim_utils

